OpenRA.Mods.Common/Activities/Air/Land.cs

```diff
@@ -42,22 +46,28 @@ public override Activity Tick(Actor self)
 					return this;
 			}
 
-			if (!target.IsValidFor(self))
+			// TargetType.Invalid indicates target was an actor and either killed, removed or teleported
+			if (target.Type == TargetType.Invalid)
+			{
+				Cancel(self);
```

Comment:
Calling `Cancel` and then immediately returning `NextActivity` looks suspicious. Shouldn't we be falling through to the `IsCanceling` check below so that we can remove the ground reservation before moving on?

Reply:
To be honest, it looks suspicious to me, too. Removed the Cancel call and merged this and the below 'if'.